solr/core/src/java/org/apache/solr/core/CoreContainer.java

```diff
@@ -2056,7 +2056,7 @@ public void reload(String name, UUID coreId) {
         newCore = core.reload(coreConfig);
 
         DocCollection docCollection = null;
-        if (getZkController() != null) {
+        if (getZkController() != null && !newCore.isSynthetic()) {
           docCollection = getZkController().getClusterState().getCollection(cd.getCollectionName());
           // turn off indexing now, before the new core is registered
           if (docCollection.getBool(ZkStateReader.READ_ONLY, false)) {
```

Comment:
perhaps this would be a little controversial since it's modifying existing behaviour for non-synthetic cores also, albeit error logic code path
```suggestion
          docCollection = getZkController().getClusterState().getCollectionOrNull(cd.getCollectionName());
          // turn off indexing now, before the new core is registered
          if (docCollection != null && docCollection.getBool(ZkStateReader.READ_ONLY, false)) {
```

Reply:
I do like your suggestion as we can avoid introduction of `isSynthetic` and SolrCore/CoreContainer can remain unaware of the synthetic collection, which is a major ➕ ! 

Perhaps we can get some input from @dsmiley on this as well? 😊  🙏🏼

Reply:
I'm good with this.

Reply:
Thanks for the input! Let's go without the `isSynthetic` then.